code/modules/mob/living/carbon/monkey/life.dm

```diff
@@ -119,151 +103,112 @@
+			if(O2_pp < safe_pressure_min)// Too little oxygen
+				if(prob(10))
+					emote("gasp")
 				if (O2_pp == 0)
 					O2_pp = 0.01
-				var/ratio = safe_oxygen_min/O2_pp
-				adjustOxyLoss(min(5*ratio, 7)) // Don't fuck them up too fast (space only does 7 after all!)
-				oxygen_alert = max(oxygen_alert, 1)
+				var/ratio = O2_pp/safe_pressure_min
+				adjustOxyLoss(min(5*ratio, CARBON_MAX_OXYLOSS))
+				oxygen_alert = TRUE
+				failed_last_breath = TRUE
 
 			else 									// We're in safe limits
-				adjustOxyLoss(-5)
-				oxygen_alert = 0
+				adjustOxyLoss(CARBON_RECOVERY_OXYLOSS)
+				oxygen_alert = FALSE
+				failed_last_breath = FALSE
 
 		if(GAS_TYPE_OXYGEN)
 			var/O2_pp = air_info[3]
-			if(O2_pp < safe_oxygen_min)// Too little oxygen
-				if(prob(20))
-					spawn(0) emote("gasp")
+			if(O2_pp < safe_pressure_min)// Too little oxygen
+				if(prob(10))
+					emote("gasp")
 				if (O2_pp == 0)
 					O2_pp = 0.01
-				var/ratio = safe_oxygen_min/O2_pp
-				adjustOxyLoss(min(5*ratio, 7)) // Don't fuck them up too fast (space only does 7 after all!)
-				oxygen_alert = max(oxygen_alert, 1)
+				var/ratio = O2_pp/safe_pressure_min
+				adjustOxyLoss(min(5*ratio, CARBON_MAX_OXYLOSS))
+				oxygen_alert = TRUE
+				failed_last_breath = TRUE
 
 			else 									// We're in safe limits
-				adjustOxyLoss(-5)
-				oxygen_alert = 0
+				adjustOxyLoss(CARBON_RECOVERY_OXYLOSS)
+				oxygen_alert = FALSE
+				failed_last_breath = FALSE
 
 		if(GAS_TYPE_N2O) //Anesthetic
 			var/SA_pp = air_info[3]
-			if(SA_pp > 20) // Enough to make us paralysed for a bit
+			if(SA_pp > 30)
+				Sleeping(10)
+			else if(SA_pp > 20) // Enough to make us paralysed for a bit
 				KnockOut(3) // 3 gives them one second to wake up and run away a bit!
 				//Enough to make us sleep as well
-				if(SA_pp > 30)
-					sleeping = min(sleeping+2, 10)
 			else if(SA_pp > 1)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
-				if(prob(20))
-					spawn(0) emote(pick("giggle", "laugh"))
+				if(prob(10))
+					emote(pick("giggle", "laugh"))
+			failed_last_breath = FALSE
 
 
 		else
-			adjustOxyLoss(7)
-			oxygen_alert = max(oxygen_alert, 1)
+			adjustOxyLoss(CARBON_MAX_OXYLOSS)
+			oxygen_alert = TRUE
+			failed_last_breath = TRUE
 
+	var/breath_temp = air_info[2]
 
-	if(air_info[2] > (T0C+66)) // Hot air hurts :(
+	if(!ISINRANGE_EX(breath_temp, BODYTEMP_COLD_DAMAGE_LIMIT_ONE, BODYTEMP_HEAT_DAMAGE_LIMIT_ONE) && !(COLD_RESISTANCE in mutations))
 		if(prob(20))
-			to_chat(src, "<span class='warning'>You feel a searing heat in your lungs!</span>")
-		fire_alert = max(fire_alert, 2)
-	else
-		fire_alert = 0
+			if(air_info[2] < BODYTEMP_COLD_DAMAGE_LIMIT_ONE)
+				to_chat(src, "<span class='danger'>You feel your face freezing and icicles forming in your lungs!</span>")
+			else
+				to_chat(src, "<span class='danger'>You feel your face burning and a searing heat in your lungs!</span>")
+
+		if(breath_temp < cold_level_3)
```

Comment:
This could fit perfectly in a `switch(breath_temp)`

Reply:
agreed